Source/core/FileSystem.h

```diff
@@ -432,7 +442,7 @@ namespace Core {
             DWORD writtenBytes;
 
             ::WriteFile(_handle, buffer, size, &writtenBytes, nullptr);
-            return static_cast<uint32_t>(writtenBytes);
```

Comment:
I think this line was oke. You are only allowing to read blocks which ara a maximum of uint32, so what you return will always be <= than 32 bits. so the static_cast was oke.

Reply:
Ok @pwielders